judgels-frontends/raphael/src/routes/contests/contests/single/submissions/Programming/ContestSubmissionsPage/ContestSubmissionsPage.tsx

```diff
@@ -59,10 +59,14 @@ export class ContestSubmissionsPage extends React.PureComponent<
     this.state = { filter: { username, problemAlias } };
   }
 
-  async componentDidMount() {
-    const { filter } = this.state;
-    if (filter.username || filter.problemAlias) {
-      await this.refreshSubmissions();
+  async componentDidUpdate(prevProps: ContestSubmissionsPageProps, prevState: ContestSubmissionsPageState) {
+    const {
+      response,
+      filter: { username, problemAlias },
+    } = this.state;
+    if (!prevState.response && response && (username || problemAlias)) {
+      this.setState({ isFilterLoading: true });
+      this.refreshSubmissions(username, problemAlias);
```

Comment:
```suggestion
      await this.refreshSubmissions(username, problemAlias);
```

Reply:
But why? Isn't the `await` will block the function call? I think it is doesn't matter to be blocking or not?

Reply:
Either add the `await`, or remove `async` from the function declaration.